CredentialProvider/CredentialProviderPresenter.swift

```diff
@@ -0,0 +1,131 @@
+                    } else {
+                        self?.cancelWith(.userInteractionRequired)
+                    }
+                }
+            }
+        }
+    }
+    
+
+    func showCredentialList(for serviceIdentifiers: [ASCredentialServiceIdentifier]) {
+    if self.profile.logins.reopenIfClosed() != nil {
+            cancelWith(.failed)
+        } else {
+            profile.logins.list().upon {[weak self] result in
+                switch result {
+                case .failure:
+                    self?.cancelWith(.failed)
+                case .success(let loginRecords):
+                    
+                    var sortedLogins = loginRecords.sorted(by: <)
+                    for (index, element) in sortedLogins.enumerated() {
+                        if let identifier = serviceIdentifiers.first?.identifier.asURL?.domainURL.absoluteString.titleFromHostname, element.passwordCredentialIdentity.serviceIdentifier.identifier.contains(identifier) {
+                            sortedLogins.remove(at: index)
+                            sortedLogins.insert(element, at: 0)
+                        }
+                    }
+                    
+                    let dataSource = sortedLogins.map { ($0.passwordCredentialIdentity, $0.passwordCredential) }
+                    DispatchQueue.main.async {
+                        self?.view?.show(itemList: dataSource)
+                    }
+                }
+            }
+        }
+    }
+    
+
+    func credentialList(for serviceIdentifiers: [ASCredentialServiceIdentifier]) {
+        view?.showWelcome()
+        
+        guard let authInfo = KeychainWrapper.sharedAppContainerKeychain.authenticationInfo(), authInfo.requiresValidation() else {
+            showCredentialList(for: serviceIdentifiers)
+            return
+        }
+        
+        AppAuthenticator.presentAuthenticationUsingInfo(
+            authInfo,
+            touchIDReason: .AuthenticationLoginsTouchReason,
+            success: { self.showCredentialList(for: serviceIdentifiers)},
+            cancel: { self.cancelWith(.userCanceled) },
+            fallback: { [weak self] in
+                self?.view?.showPassword { isOk in
+                    if isOk { self?.showCredentialList(for: serviceIdentifiers) }
+                }
+            })
+    }
+    
+
+    func prepareAuthentication(for credentialIdentity: ASPasswordCredentialIdentity) { }
+}
+
+@available(iOS 12, *)
+private extension CredentialProviderPresenter {
+    
+    func displayNotLoggedInMessage() {
+        view?.displayAlertController(
+            buttons: [
+                AlertActionButtonConfiguration(
+                    title: "OK",
+                    tapAction: { [weak self] in self?.cancelWith(.userCanceled) },
+                    style: .default)
+            ],
+            title: "NOt signed in",
```

Comment:
Is this a dialog we need to localize? Does this need strings?

Reply:
I have removed this whole function and simplified `extensionConfigurationRequested()`, which now does the following:

```swift
    func extensionConfigurationRequested() {
        view?.showWelcome()
        if self.profile.logins.reopenIfClosed() != nil {
            // At this point there is nothing useful we can do if we cannot open the logins database. Worst case
            // we skip the synchronization and not all logins will be available to the user if they have changed
            // since the last time.
            return
        }
        
        profile.syncCredentialIdentities().upon { result in
            sleep(2)
            self.cancelWith(.userCanceled)
        }
    }
```

Previously it would call `displayNotLoggedInMessage()` if `reopenIfClosed()` would fail but I explained in the above code comment why I think that was not the best path.

Reply:
cc @razvanlitianu for visibility on this change.